hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -488,7 +488,7 @@ private int getBlockCount(List<ExtendedBlocklet> blocklets) {
     // This is to log the event, so user will know what is happening by seeing logs.
     LOG.info("Started block pruning ...");
     if (carbonTable.isTransactionalTable()) {
-      prunedBlocklets = defaultDataMap.prune(segmentIds, resolver, partitionsToPrune);
+      prunedBlocklets = defaultDataMap.prune(segmentIds, expression, resolver, partitionsToPrune);
```

Comment:
Better wrap `expression` and `resolver` to one object and send it to the tabledatamap. 
And also remove the else part method as all will fall to the same method.